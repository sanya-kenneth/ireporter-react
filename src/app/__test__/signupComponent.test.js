import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SignupComponent from '../components/signupComponent';

describe('<SignupComponent/>', () => {
  let wrapper;
  const PropsTest = {
    signupResponse: {},
    signupErrors: {},
    handleChange: jest.fn(),
    handleSignup: jest.fn(),
    isLoading: false,
  };
  it('renders signup page', () => {
    wrapper = shallow(<SignupComponent {...PropsTest} />);
  });
  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
