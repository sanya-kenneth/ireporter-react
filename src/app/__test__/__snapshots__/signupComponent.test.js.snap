// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupComponent/> matches snapshot 1`] = `
<Fragment>
  <div
    className="signup-back"
  >
    <div
      className="row"
    >
      <div
        className="col-md-4"
      />
      <div
        className="col-md-4"
      >
        <img
          className="logo"
          src="../../../src/assets/images/logo.png"
        />
        <div
          className="card signup-card"
        >
          <form
            onSubmit={[MockFunction]}
          >
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="firstname"
                onChange={[MockFunction]}
                placeholder="firstname"
                type="text"
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="lastname"
                onChange={[MockFunction]}
                placeholder="lastname"
                type="text"
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="othernames"
                onChange={[MockFunction]}
                placeholder="othernames"
                type="text"
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="email"
                onChange={[MockFunction]}
                placeholder="email"
                type="email"
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="username"
                onChange={[MockFunction]}
                placeholder="username"
                type="text"
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="phonenumber"
                onChange={[MockFunction]}
                placeholder="phonenumber"
                type="number"
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="password"
                onChange={[MockFunction]}
                placeholder="password"
                type="password"
              />
            </div>
            <button
              className="btn btn-primary"
            >
              Signup
            </button>
            <p>
              Already have an account!
              <Link
                to="/"
              >
                Login
              </Link>
            </p>
          </form>
        </div>
      </div>
      <div
        className="col-md-4"
      />
    </div>
  </div>
</Fragment>
`;
